src/draw/DefaultDrawer.js

```diff
@@ -106,466 +160,464 @@ class DefaultDrawer {
+      })
+      .round(true);
+
+    const rootNode = d3.hierarchy(this.shadowRoot);
+
+    rootNode
+      .count()
+      .sort((a, b) => b.height - a.height
+            || b.value - a.value
+            || (b.data.definition?.isContainer ?? 0)
+            - (a.data.definition?.isContainer ?? 0));
+
+    treemapLayout(rootNode);
+
+    return d3.groups(
+      rootNode,
+      (data) => (data.parent
+        && data.parent.data.id !== '__shadowRoot'
+        ? `group-${data.parent.data.id}`
+        : 'root-components'),
+    )
+      .filter((data) => data[0] !== 'root-__shadowRoot');
+  }
+
+  /**
+   * Get the anchor point on source selection facing target selection
+   * @param {Selection} sourceSelection - The source D3 selection object
+   * @param {Selection} targetSelection - The target D3 selection object
+   * @returns {[Number,Number] || null} - Tuple representing x,y coordinates,
+   * null if lacking source and/or target
+   */
+  getAnchorPoint(sourceSelection, targetSelection) {
+    if (sourceSelection.empty() || targetSelection.empty()) {
+      return null;
     }
-  }
 
-  /**
-   * Add a link between two components
-   * @param {ComponentLinkDefinition[]} linkDefinitions - Array that contains all link definitions
-   * allowed.
-   * @private
-   */
-  __addLink(linkDefinitions) {
-    const source = this.actions.selection.current;
-    const targetSelector = this.actions.link.targetList.map((type) => `.${type}`).join(', ');
-    if (targetSelector === '') {
-      return;
+    const sourceCoords = sourceSelection.node().getBoundingClientRect();
+    const targetCoords = targetSelection.node().getBoundingClientRect();
+    const anchorPoint = { x: sourceCoords.x, y: sourceCoords.top + (sourceCoords.height / 2) };
+
+    if (targetCoords.right < sourceCoords.left) {
+      anchorPoint.x = sourceCoords.left;
+    } else {
+      anchorPoint.x = sourceCoords.right;
     }
 
-    this.interact(targetSelector).on(
-      'click',
-      (event) => {
-        if (source !== event.currentTarget.id) {
-          const target = this.d3.select(`#${event.currentTarget.id}`).datum();
-          const link = new ComponentLink({
-            source,
-            target: event.currentTarget.id,
-            definition: linkDefinitions
-              .find((def) => def.targetRef === target.definition.type),
-          });
-          const existingLink = this.pluginData.links
-            .filter((item) => item.source === link.source && item.target === link.target);
-
-          if (!existingLink.length) {
-            this.pluginData.links.push(link);
-            this.d3.select(`#${source}`).datum().setLinkAttribute(link);
-            this.drawLinks();
-          }
-        }
-        this.interact(targetSelector).unset();
-        this.d3.selectAll('.component').style('opacity', 1);
-        this.actions.link.targetList = [];
-      },
-    );
-  }
+    const transformedCoordinates = this.screenToSVG(anchorPoint.x, anchorPoint.y);
```

Comment:
please destructure your object in : `const {x, y} = this.screentoSVG(...)`

Reply:
done